From 6629a951d4cb8c1b6cf53ae9863a66b9e4c920aa Mon Sep 17 00:00:00 2001
Message-ID: <6629a951d4cb8c1b6cf53ae9863a66b9e4c920aa.1731584860.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1731584860.git.sharlatanus@gmail.com>
References: <cover.1731584860.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 4 Nov 2024 22:21:35 +0000
Subject: [PATCH v2 02/14] gnu: python-s3transfer: Update to 0.10.3.

* gnu/packages/python-web.scm (python-s3transfer): Update to 0.10.3.
[build-system]: Swap to pyprojct-build-system.
[native-inputs]: Add python-pytest, python-pytest-xdist,
python-setuptools, and python-wheel.

Change-Id: I41d35dc89ea6f9ff24f33f6bfcc1fc6a22ebe890
---
 gnu/packages/python-web.scm | 46 ++++++++++++++++++++-----------------
 1 file changed, 25 insertions(+), 21 deletions(-)

diff --git a/gnu/packages/python-web.scm b/gnu/packages/python-web.scm
index 7bbf05f368..b18248e472 100644
--- a/gnu/packages/python-web.scm
+++ b/gnu/packages/python-web.scm
@@ -4718,31 +4718,35 @@ (define-public python-betamax-serializers
 (define-public python-s3transfer
   (package
     (name "python-s3transfer")
-    (version "0.5.0")
-    (source (origin
-              (method url-fetch)
-              (uri (pypi-uri "s3transfer" version))
-              (sha256
-               (base32
-                "0k6sc956yrrv9b4laa0r79jhxajpyxr21jcd1ka8m1n53lz85vah"))))
-    (build-system python-build-system)
+    (version "0.10.3")
+    (source
+     (origin
+       (method url-fetch)
+       (uri (pypi-uri "s3transfer" version))
+       (sha256
+        (base32 "032bjky1q8r5x80mvb0ah60g0zq4snwf0xa4c7779m44mdsfsl2g"))))
+    (build-system pyproject-build-system)
     (arguments
-     `(#:phases
-       (modify-phases %standard-phases
-         (replace 'check
-           (lambda* (#:key tests? #:allow-other-keys)
-             (when tests?
-               ;; Some of the 'integration' tests require network access or
-               ;; login credentials.
-               (invoke "nosetests" "--exclude=integration")))))))
+     (list
+      #:test-flags
+      #~(list "--numprocesses" "auto"
+              ;; Tests require networking.
+              "--ignore" "tests/integration")))
     (native-inputs
-     (list python-docutils python-mock python-nose))
+     (list python-docutils
+           python-mock
+           python-nose
+           python-pytest
+           python-pytest-xdist
+           python-setuptools
+           python-wheel))
     (propagated-inputs
-     (list python-botocore python-urllib3))
-    (synopsis "Amazon S3 Transfer Manager")
-    (description "S3transfer is a Python library for managing Amazon S3
-transfers.")
+     (list python-botocore
+           python-urllib3))
     (home-page "https://github.com/boto/s3transfer")
+    (synopsis "Amazon S3 Transfer Manager")
+    (description
+     "S3transfer is a Python library for managing Amazon S3 transfers.")
     (license license:asl2.0)))
 
 (define-public python-flask-jwt
-- 
2.46.0

